cpp/src/slice2cs/Gen.cpp

```diff
@@ -2880,23 +2875,18 @@ Slice::Gen::DispatcherVisitor::writeOutgoingResponseWriter(const OperationPtr& o
     string ns = getNamespace(interface);
 
     auto returns = operation->returnValues();
+    assert(!returns.empty());
 
     bool defaultWriter = returns.size() == 1 && operation->returnBitSequenceSize() == 0 && !returns.front()->tagged();
     if (defaultWriter)
     {
+        // This includes operations with a single struct return value.
         _out << outputStreamWriter(returns.front()->type(), ns, false);
     }
-    else if (returns.size() > 1)
-    {
-        _out << "(ZeroC.Ice.OutputStream ostr, in " << toTupleType(returns, true) << " value) =>";
-        _out << sb;
-        writeMarshal(operation, true);
-        _out << eb;
-    }
     else
     {
-        auto p = returns.front();
-        _out << "(ZeroC.Ice.OutputStream ostr, " << paramTypeStr(p, true) << " value) =>";
+        _out << "(ZeroC.Ice.OutputStream ostr, " << (returns.size() > 1 ? "in " : "") << toTupleType(returns, true)
```

Comment:
Shouldn't we also use `in` when the return  type is a struct?

Reply:
See the comments above in the code. The single-struct return is handled by the "defaultWriter" block.